### ISSUE TITLE
Sidenav causes an exception in angular-animate

### ISSUE THREAD

**Issue by oliverzy** [angular/material#2261]
Today I updated to Angular 1.4 RC0, I found an exception is thrown in this line:
https://github.com/angular/material/blame/master/src/components/sidenav/sidenav.js#L292

After looking at the document, $animate.enter and $animate.leave has different signature, I changed the line to ` isOpen ? $animate.enter(backdrop, parent) : $animate.leave(backdrop),`, the problem solved.

**Comment by ThomasBurleson** [comment]
@oliverzy - Great catch.
Yes both [Angular 1.3.x $animate](https://code.angularjs.org/1.3.14/docs/api/ng/service/$animate) and [1.4.x $animate](https://code.angularjs.org/1.4.0-rc.0/docs/api/ng/service/$animate) have the same API:

*  `$animate.enter(element, parent, [after], [options]);`
*  `$animate.leave(element, [options]);`

The mdSideNav code is simply wrong.

**Comment by phyllisstein** [comment]
This may not be altogether germane, but it looks like some of the other changes to `$animate` in Angular 1.4.0-rc.0 may be causing further problems: after your change, the sidenav no longer throws an error but it doesn't exactly animate either---it just pops in and out. Downgrading to 1.4.0-beta.6 restores the animation.

**Comment by ThomasBurleson** [comment]
@phyllisstein - The [Online Docs SideNav](https://material.angularjs.org/#/demo/material.components.sidenav) uses the `$animate` changes and still animates. Please provide a CodePen or Plunkr that demonstrates your issue.

**Comment by phyllisstein** [comment]
@ThomasBurleson Right you are! Couldn't reproduce it; must be something funky in my application. Sorry for the bother!

**Comment by phyllisstein** [comment]
@ThomasBurleson On second thought, I was able to reproduce it here, in Safari and Chrome: <http://plnkr.co/edit/jzI6LmIWGwFPBzjJ17y3?p=preview>. The issue seems to occur only when I bootstrap the application manually, with `angular.bootstrap`; if I set an `ng-app` on the document, it works as expected.

**Comment by ThomasBurleson** [comment]
Interesting...

**Comment by ThomasBurleson** [comment]
@matsko - any idea why this $animate issue is occurring with a bootstrap process ?

**Comment by ThomasBurleson** [comment]
@matsko  - Here is a revised CodePen [SideNav with Material and Angular.js 1.4.x](http://plnkr.co/edit/XagFuIpE9loOAqtqplQX?p=preview) that manifests this:

![screen shot 2015-04-12 at 11 54 39 am](https://cloud.githubusercontent.com/assets/210413/7106575/36311506-e10b-11e4-8b08-80a19d5f042b.png)

The interesting issue is that the SideNav animation still works if you are not bootstrapping (and the above error still occurs). But if you boot strap:

```html
<script type="text/javascript">
      angular.element(document).ready(function() {
        angular.bootstrap(document, ['sidenav']);
      });
    </script>
```

Then the slidein animation no longer works.

**Comment by matsko** [comment]
@ThomasBurleson I'm not seeing the error in the provided plunkr. Did you intend to use another link?

**Comment by ThomasBurleson** [comment]
@matsko  - I do not see the type error now
The Plunkr is now updated to use angular.bootstrap().... notice the slideIn effect is not used and the SideNav jumps into view without animations.

**Comment by matsko** [comment]
If you do:

```js
angular.bootstrap(document.body, ['sidenav']);
angular.bootstrap(document.querySelector('html'), ['sidenav']);
```

Then it works. Perhaps using `document` directly has issues with the new ngAnimate code. I'll investigate.

**Comment by ThomasBurleson** [comment]
Closing this issue.

**Comment by matsko** [comment]
@ThomasBurleson this is a legitimate issue. I've created an issue and a pull request to fix it: https://github.com/angular/angular.js/issues/11574 and https://github.com/angular/angular.js/pull/11575

**Comment by ThomasBurleson** [comment]
@matsko - thx for the super fast turnaround!

**Comment by matsko** [comment]
This has now been merged and fixed in master for 1.4. https://github.com/angular/angular.js/commit/bee14ed1e7b77ea7dc62326611380da36dec297e

**Comment by ThomasBurleson** [comment]
@matsko - awesome. thx for the follow-up.
